packages/jss/src/plugins/rules.js

```diff
@@ -30,4 +31,27 @@ const plugins: $ReadOnlyArray<Plugin> = Object.keys(classes).map((key: string) =
   return {onCreateRule}
 })
 
+// Animation name ref replacer.
+plugins.push({
+  onProcessStyle: (style, rule, sheet) => {
+    if (rule.type !== 'style' || !sheet) return style
+
+    // We need to support camel case here, because this plugin runs before the camelization plugin.
+    const prop = 'animationName' in style ? 'animationName' : 'animation-name'
```

Comment:
I think we also need to support `animation: { name: 'fadeIn' }`

Reply:
and also the `animation: $keyframeId 150ms` shortcut

Reply:
thought about it an decided to limit it for now, because it goes into difficult directions, adds more code and slows down.

In the first case you would introduce the knowledge from jss-expand plugin about new syntax, in the second case, you would have to always run a regex or indexOf on the animation property.

We already had to introduce camel case support in the core, which is suboptimal. In theory we could make this refs replacement a separate plugin, but then its not very logical that one can't reference keyframes from the same sheet without plugins.

Reply:
Making it a separate plugin and running after jss-expand would allow to support "animationName", because jss-expand can generate it

Reply:
Another thought I had a while ago: what if we extract all core rules into separate packages and plugins? Most people use preset anyways. A few people can also setup by themselves all the plugins. This would give us the flexibility with above mentioned plugin.

Reply:
Additional benefit: someone who doesn't need all the rules, doesn't have to include them. For e.g. we got a request to include @page rule and there are some more rarely used rules. Why not make them all separate packages, but included in default preset.

Reply:
Yeah, let's maybe make all of the rules plugins for now and then move them out maybe.

We could do a `KeyframeRulePlugin` and a `KeyframeReferencePlugin` then. I think this would be the best solution. 

This should allow us to support all of the mentioned case above without needing to introduce anything.

Reply:
by "mak all of the rules plugins" you mean just jss/src/plugins or making them separate packages each?

Reply:
lets do this in  a separate PR after the other PRs are merged, too many conflicts otherwise

Reply:
Okay, I think we need to apply the rule plugins last so that the keyframes name replace works on every possibility.